pyiceberg/catalog/hive.py

```diff
@@ -271,6 +271,18 @@ def __init__(self, name: str, **properties: str):
             DEFAULT_LOCK_CHECK_RETRIES,
         )
 
+    @staticmethod
+    def _create_hive_client(properties: Dict[str, str]) -> _HiveClient:
+        uris = properties["uri"].split(",")
+        for idx, uri in enumerate(uris):
+            try:
+                return _HiveClient(uri, properties.get("ugi"))
+            except BaseException as e:
+                if idx + 1 == len(uris):
+                    raise e
+                else:
+                    continue
```

Comment:
Can I make a suggestion to make it a bit more Pythonic (and easier to follow IMHO):
```suggestion
    @staticmethod
    def _create_hive_client(properties: Dict[str, str]) -> _HiveClient:
        uris = properties["uri"]
        e = ValueError(f"Invalid URI: {uris}")
        for uri in uris.split(","):
            try:
                return _HiveClient(uri, properties.get("ugi"))
            except BaseException as e:
                pass
        raise e
```

Reply:
Rather than throwing a generic `ValueError` I have updated to capture and throw the last exception if that code path is reached. Let me know what you think